File: src/main/java/com/randomlychosenbytes/jlocker/nonabstractreps/SuperUser.java
package com.randomlychosenbytes.jlocker.nonabstractreps;

import com.google.gson.annotations.Expose;
import com.randomlychosenbytes.jlocker.manager.Utils;

import javax.crypto.SecretKey;

public class SuperUser extends User {

    @Expose
    protected String encSuperUMasterKeyBase64;

    public SuperUser(
            String password,
            String hash,
            String encUserMasterKeyBase64,
            String encSuperUMasterKeyBase64
    ) {
        super(password);

        this.passwordHash = hash;
        this.encSuperUMasterKeyBase64 = encSuperUMasterKeyBase64;
        this.encryptedUserMasterKeyBase64 = encUserMasterKeyBase64;
    }

    public SecretKey getSuperUMasterKeyBase64() {
        return Utils.decryptKeyWithString(encSuperUMasterKeyBase64, decryptedUserPassword);
    }
}

File: src/main/java/com/randomlychosenbytes/jlocker/model/SuperUser.kt
package com.randomlychosenbytes.jlocker.model

import com.google.gson.annotations.Expose
import com.randomlychosenbytes.jlocker.utils.encryptKeyWithString
import com.randomlychosenbytes.jlocker.utils.generateKey

class SuperUser(password: String) : User(
    password, encryptKeyWithString(generateKey(), password)
) {
    @Expose
    val encryptedSuperUMasterKeyBase64: String = encryptKeyWithString(generateKey(), password)
}